pkg/security/rconfig/profiles.go

```diff
@@ -0,0 +1,147 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package rconfig
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	proto "github.com/DataDog/agent-payload/v5/cws/dumpsv1"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/data"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	cgroupModel "github.com/DataDog/datadog-agent/pkg/security/resolvers/cgroup/model"
+	"github.com/DataDog/datadog-agent/pkg/security/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// image name/image tag separator
+	separator = ":::"
+)
+
+type ProfileConfig struct {
+	Tags    []string
+	Profile []byte
+}
+
+type RCProfileProvider struct {
+	sync.RWMutex
+
+	client *remote.Client
+
+	onNewProfileCallback func(selector cgroupModel.WorkloadSelector, profile *proto.SecurityProfile)
+}
+
+// Close stops the client
+func (r *RCProfileProvider) Stop() error {
+	r.client.Close()
+	return nil
+}
+
+func (r *RCProfileProvider) rcProfilesUpdateCallback(configs map[string]state.ConfigCWSProfiles) {
+	for _, config := range configs {
+		var profCfg ProfileConfig
+		if err := json.Unmarshal(config.Config, &profCfg); err != nil {
+			log.Errorf("couldn't decode json profile: %s", err)
+			continue
+		}
+
+		profile := &proto.SecurityProfile{}
+		if err := profile.UnmarshalVT([]byte(profCfg.Profile)); err != nil {
+			log.Errorf("couldn't decode protobuf profile: %s", err)
+			continue
+		}
+
+		imageName := utils.GetTagValue("image_name", profile.Tags)
+		imageTag := utils.GetTagValue("image_tag", profile.Tags)
+
+		if imageName == "" {
+			log.Errorf("no image name: %v", profile.Tags)
+			continue
+		}
+
+		if imageTag == "" {
+			imageTag = "latest"
+			profile.Tags = append(profile.Tags, "image_tag:"+imageTag)
+		}
```

Comment:
This logic is more or less already in `NewWorkloadSelector`, should we merge the 2 to make sure we handle the image name and tag consistently ?